### PR TITLE
Test storage and provider-id via integration testing

### PR DESCRIPTION
**Pull request by addyess**
Adding integration test to test storage and presence of provider-ids

### CODE REVIEW

File: tests/data/bind_pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      storage: 100Mi
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: busybox
+  namespace: default
+spec:
+  containers:
+    - image: busybox
```
Comment (kwmonroe) [inline tests/data/bind_pvc.yaml]:
```suggestion
    - image: rocks.canonical.com/cdk/busybox:1.36
```

File: tests/data/charm.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+description: Overlay for attaching current charm
+series: {{ series }}
+applications:
+  aws-integrator:
+    charm: {{ charm }}
+    channel: null
+    resources:
+      aws-cli: 0
+    num_units: 1
+    trust: True
```
Comment (kwmonroe) [inline tests/data/charm.yaml]:
```suggestion
    trust: true
```

It'll work with `T`, but lc is consistent with our other test data and overlays.

**Review by kwmonroe — APPROVED** [review]
minor yaml tweaks for your consideration. i also swapped out `juju scp` for `juju ssh` in a separate commit for j3 compat.

otherwise LGTM as a nice addition for manual integration runs.  i'm also +1 to include in the gh workflow if you'd like.

**Comment by addyess** [comment]
> i'm also +1 to include in the gh workflow if you'd like.

I don't believe we had AWS secrets creds in our organization to launch on AWS from github

**Pull request merged**